pallets/pips/src/lib.rs

```diff
@@ -1181,13 +1202,22 @@ decl_module! {
             })
         }
 
-        /// When constructing a block check if it's time for a ballot to end. If ballot ends,
-        /// proceed to ratification process.
-        fn on_initialize(n: T::BlockNumber) -> Weight {
-            Self::end_block(n).unwrap_or_else(|e| {
-                sp_runtime::print(e);
-                0
-            })
+        /// Internal dispatchable that handles execution of a PIP.
+        #[weight = (1_000_000_000, DispatchClass::Operational, Pays::Yes)]
+        fn execute_scheduled_pip(origin, id: PipId) -> DispatchResultWithPostInfo {
```

Comment:
Is there anyway to make this function and `expire_scheduled_pip` only callable by the Scheduler?

Could you remind me of the reason we need to schedule this function, rather than directly scheduling the underlying PIP?

Reply:
As per discussion, we can return an Err from execute_proposal if the states etc. don't line up.